tests/app/dao/test_annual_billing_dao.py

```diff
@@ -0,0 +1,35 @@
+from app.service.utils import get_current_financial_year_start_year
```

Comment:
It doesn't look like you are explicitly testing all the methods from the dao

e.g. dao_get_annual_billing
dao_create_or_update_annual_billing_for_year
dao_get_free_sms_fragment_limit_for_year
dao_get_all_free_sms_fragment_limit

also the insert mwthod which is used externally to the dao class.

Reply:
The insert method is in tested in rest_rest.py test_create_service_free_sms_fragment_limit_is_optional()
New test methods are added.

Reply:
I'd be inclined to test it separately here. rest test are like integration/system tests. 

Technically the REST method could be deleted and it could be used in another place (in future).